src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -577,63 +579,113 @@ def create_or_update_test_run_body(
     return new_body
 
 
-def upload_files_helper(
-    client, test_id, yaml_data, test_plan, load_test_config_file, wait
+def upload_properties_file_helper(
+    client, test_id, yaml_data, load_test_config_file, existing_test_files, wait
 ):
-    files = client.list_test_files(test_id)
-    if yaml_data:
+    if yaml_data and yaml_data.get("properties", {}).get("userPropertyFile") is not None:
         user_prop_file = yaml_data.get("properties", {}).get("userPropertyFile")
-        if user_prop_file is not None:
-            logger.info("Uploading user property file %s", user_prop_file)
-            file_name = os.path.basename(user_prop_file)
-            for file in files:
-                if AllowedFileTypes.USER_PROPERTIES.value == file["fileType"]:
-                    client.delete_test_file(test_id, file["fileName"])
-                    logger.info(
-                        "File of type '%s' already exists in test %s. Deleting it!",
-                        AllowedFileTypes.USER_PROPERTIES,
-                        test_id,
-                    )
-                    break
-            file_response = upload_file_to_test(
-                client,
-                test_id,
-                user_prop_file,
-                file_type=AllowedFileTypes.USER_PROPERTIES,
-                wait=wait,
-            )
-            logger.info(
-                "Uploaded file '%s' of type %s to test %s",
-                file_name,
-                AllowedFileTypes.USER_PROPERTIES,
-                test_id,
+        if not os.path.isabs(user_prop_file) and load_test_config_file:
+            yaml_dir = os.path.dirname(load_test_config_file)
+            user_prop_file = os.path.join(yaml_dir, user_prop_file)
+        logger.info("Uploading user property file %s", user_prop_file)
+        file_name = os.path.basename(user_prop_file)
+        for file in existing_test_files:
+            if AllowedFileTypes.USER_PROPERTIES.value == file["fileType"]:
+                client.delete_test_file(test_id, file["fileName"])
+                logger.info(
+                    "File of type '%s' already exists in test %s. Deleting it!",
+                    AllowedFileTypes.USER_PROPERTIES,
+                    test_id,
+                )
+                break
+        file_response = upload_file_to_test(
+            client,
+            test_id,
+            user_prop_file,
+            file_type=AllowedFileTypes.USER_PROPERTIES,
+            wait=wait,
+        )
+        logger.info(
+            "Uploaded file '%s' of type %s to test %s",
+            file_name,
+            AllowedFileTypes.USER_PROPERTIES,
+            test_id,
+        )
+        if wait and file_response.get("validationStatus") != "VALIDATION_SUCCESS":
+            raise FileOperationError(
+                f"Properties file {file_name} is not valid. Please check the file and try again."
```

Comment:
The deletion logic is slightly different in case of user_properties and test_plan and test_plan is a different object in itself, so didn't use the upload_generic_files_helper for these two because handling these two differently increases the complexity of the method and we'll be back to square one
![image](https://github.com/user-attachments/assets/8393b55f-7220-45d0-ba21-1e613edcf568)

Reply:
Got it, was trying to reduce code duplication. See if there's some possibility around here

Reply:
refactored a bit by defining what `existing_files` constitute for the various file types